app/controllers/security_group_controller.rb

```diff
@@ -14,4 +14,250 @@ def self.display_methods
   end
 
   menu_section :net
+
+  def button
+    @edit = session[:edit] # Restore @edit for adv search box
+    params[:display] = @display if %w(vms instances images).include?(@display)
+    params[:page] = @current_page unless @current_page.nil? # Save current page for list refresh
+
+    @refresh_div = "main_div"
+
+    case params[:pressed]
+    when "security_group_tag"
+      return tag("SecurityGroup")
+    when 'security_group_delete'
+      delete_security_groups
+    when "security_group_edit"
+      checked_security_group_id = get_checked_security_group_id(params)
+      javascript_redirect :action => "edit", :id => checked_security_group_id
+    else
+      if params[:pressed] == "security_group_new"
+        javascript_redirect :action => "new"
+      elsif !flash_errors? && @refresh_div == "main_div" && @lastaction == "show_list"
+        replace_gtl_main_div
+      else
+        render_flash
+      end
+    end
+  end
+
+  def cancel_action(message)
+    session[:edit] = nil
+    @breadcrumbs.pop if @breadcrumbs
+    javascript_redirect :action    => @lastaction,
+                        :id        => @security_group.id,
+                        :display   => session[:security_group_display],
+                        :flash_msg => message
+  end
+
+  def security_group_form_fields
+    assert_privileges("security_group_edit")
+    security_group = find_by_id_filtered(SecurityGroup, params[:id])
+    render :json => {
+      :name              => security_group.name,
+      :description       => security_group.description,
+      :cloud_tenant_name => security_group.cloud_tenant.try(name),
```

Comment:
Name without the colon is a variable or a method call. You really need to pass a symbol to try. Please fix this.